matter_server/client/connection.py

```diff
@@ -0,0 +1,166 @@
+"""Logic to manage the WebSocket connection to the Matter server."""
+from __future__ import annotations
+
+import asyncio
+import logging
+import pprint
+from typing import Any, Callable, Dict, Final, cast
+
+from aiohttp import ClientSession, ClientWebSocketResponse, WSMsgType, client_exceptions
+
+from ..common.helpers.json import json_dumps, json_loads
+from ..common.helpers.util import chip_clusters_version, parse_message
+from ..common.models.events import EventType
+from ..common.models.message import CommandMessage, MessageType, ServerInfoMessage
+from ..common.models.node import MatterNode
+from ..common.models.server_information import ServerInfo
+from .const import MIN_SCHEMA_VERSION
+from .exceptions import (
+    CannotConnect,
+    ConnectionClosed,
+    ConnectionFailed,
+    InvalidMessage,
+    InvalidServerVersion,
+    InvalidState,
+    NotConnected,
+)
+
+LOGGER = logging.getLogger(f"{__package__}.connection")
+SUB_WILDCARD: Final = "*"
+
+
+class MatterClientConnection:
+    """Manage a Matter server over WebSockets."""
+
+    def __init__(
+        self,
+        ws_server_url: str,
+        aiohttp_session: ClientSession,
+    ):
+        """Initialize the Client class."""
+        self.ws_server_url = ws_server_url
+        self.aiohttp_session = aiohttp_session
```

Comment:
I'm wondering if we should make the session protected. It's not meant for anything else to use, right?

Reply:
yeah, good idea, I have no clue why you would want to make it public

Reply:
adjusted it